test/boost/sstable_move_test.cc

```diff
@@ -35,14 +35,15 @@ SEASTAR_THREAD_TEST_CASE(test_sstable_move) {
     auto env = test_env();
     auto stop_env = defer([&env] { env.stop().get(); });
 
-    sstables::generation_type::int_t gen = 1;
-    auto [ sst, cur_dir ] = copy_sst_to_tmpdir(tmp.path(), env, uncompressed_schema(), fs::path(uncompressed_dir()), gen);
+    sstables::sstable_generation_generator gen_generator{0};
```

Comment:
To further detach tests from concrete values, we could add a default value for `sstable_generation_generator`, which chooses an appropriate start value for itself (which will practically still be 0).

Reply:
explained above. let's do this in smaller steps.